packages/solid/store/src/store.ts

```diff
@@ -11,28 +11,8 @@ export type StoreNode = {
   [k: number]: any;
 };
 
-// well-known symbols need special treatment until https://github.com/microsoft/TypeScript/issues/24622 is implemented.
-type AddSymbolToPrimitive<T> = T extends { [Symbol.toPrimitive]: infer V }
-  ? { [Symbol.toPrimitive]: V }
-  : {};
-type AddSymbolIterator<T> = T extends { [Symbol.iterator]: infer V }
-  ? { [Symbol.iterator]: V }
-  : {};
-type AddSymbolToStringTag<T> = T extends { [Symbol.toStringTag]: infer V }
-  ? { [Symbol.toStringTag]: V }
-  : {};
-type AddCallable<T> = T extends { (...x: any[]): infer V } ? { (...x: Parameters<T>): V } : {};
-
-export type NotWrappable = string | number | boolean | Function | null;
-// Intersection for missing fields https://github.com/microsoft/TypeScript/issues/13543
-export type Store<T> = {
-  [P in keyof T]: T[P] extends object ? Store<T[P]> & T[P] : T[P];
-} & {
-  [$RAW]?: T;
-} & AddSymbolToPrimitive<T> &
-  AddSymbolIterator<T> &
-  AddSymbolToStringTag<T> &
-  AddCallable<T>;
+export type NotWrappable = string | number | bigint | boolean | Function | null;
```

Comment:
Shouldn't (Weak)?(Map|Set) also be on that list? I know they are rarely used, but I don't think we'll map them to a proxy in a store.

Reply:
I think all classes are on that list, but I'm not sure how to type "any class instance". Though I think there was some talk on discord about allowing user classes to be wrappable somehow. That said, listing all standard classes doesn't seem like the right way to go.

Reply:
So it turns out if I define `type Wrappable = readonly unknown[] | Record<string, unknown>` and use it instead of `NotWrappable`, nothing breaks and only plain objects and arrays are allowed. Two errors appear [here](https://github.com/solidjs/solid/blob/main/packages/solid/store/test/store.spec.ts#L478):
```ts
    setStore("inner", "a", 10);
    //                ^ string is not assignable to never
    expect(sum).toBe(11);
    setStore("inner", "b", 5);
    //                ^ string is not assignable to never
    expect(sum).toBe(11);
```
...but since these setStore calls don't trigger updates, the error is possibly desirable? 

IThere might also be a slight naming issue here: `wrap` is used to wrap more than `Wrappable` and what passes `isWrappable`. Not sure if that will cause any confusion later on.